packages/ringcentral-integration/modules/ConferenceCall/index.js

```diff
@@ -796,71 +794,49 @@ export default class ConferenceCall extends RcModule {
   }
 
   @proxify
-  async onMerge({ sessionId }) {
-    const session = this._webphone._sessions.get(sessionId);
-    const isSessionOnhold = session.isOnHold().local;
-    this.setMergeParty({ toSessionId: sessionId });
-    const sessionToMergeWith = this._webphone._sessions.get(this.mergingPair.fromSessionId);
+  async mergeSession({ sessionId }) {
+    const session = find(
+      x => x.id === sessionId,
+      this._webphone.sessions
+    );
+
+    if (this._webphone.isCallRecording(session)) {
+      return null;
+    }
+
+    this.setMergeParty({
+      toSessionId: sessionId,
+    });
+
+    const sessionToMergeWith = find(
+      x => x.id === this.mergingPair.fromSessionId,
+      this._webphone.sessions
+    );
     const webphoneSessions = sessionToMergeWith
       ? [sessionToMergeWith, session]
       : [session];
     await this.mergeToConference(webphoneSessions);
+
     const conferenceData = Object.values(this.conferences)[0];
     if (!conferenceData) {
       await this._webphone.resume(session.id);
       return null;
     }
 
-    if (isSessionOnhold) {
+    if (session.isOnHold) {
```

Comment:
And according to the requirement [RCINT-7771](https://jira.ringcentral.com/browse/RCINT-7711),  the final holding/unholding status is depend on the current call.